src/kakarot/instructions.cairo

```diff
@@ -612,8 +613,16 @@ namespace EVMInstructions {
             if (is_parent_root != FALSE) {
                 return ctx;
             } else {
-                let ctx = CallHelper.finalize_calling_context(ctx);
-                return run(ctx=ctx);
+                let (bytecode_len) = IEvmContract.bytecode_len(
+                    contract_address=ctx.starknet_contract_address
+                );
+                if (bytecode_len == 0) {
+                    let ctx = CreateHelper.finalize_calling_context(ctx);
```

Comment:
why do we need this if/else?

Reply:
the CALLs & the CREATEs opcodes both require a finalize when they RETURN, but they don't require the same: CALLs push `success` while `CREATEs push `address`.

Reply:
but here the code is identical, no?

Reply:
it's `CreateHelper.{}` and `CallHelper.{}`